sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala

```diff
@@ -301,7 +301,6 @@ class DataFrameReaderWriterSuite extends QueryTest with SharedSQLContext with Be
       intercept[AnalysisException] {
         spark.range(10).write.format("csv").mode("overwrite").partitionBy("id").save(path)
       }
-      spark.emptyDataFrame.write.format("parquet").mode("overwrite").save(path)
     }
```

Comment:
Its not legal to write an empty struct in parquet.  Its explained by Herman in [SPARK-20593](https://issues.apache.org/jira/browse/SPARK-20593). Previously, we didn't setup a write 
task for this where as now with this fix we do.